### DISCUSSION TITLE
Conditional formatting improvements

### DISCUSSION THREAD

**Discussion by Gradsta** [apache/superset#22790] [Ideas]
I didn't see these suggested in issues or discussions so hopefully these aren't duplicates. I recently tried to apply conditional formatting to a table in Superset and found myself longing for the following:

1. The ability to apply conditional formatting to non-numeric columns (specifically text and boolean columns).
2. The ability to toggle highlighting for an entire row instead of just an individual cell.

**Comment by gmilman** [comment]
I would find these developments very useful too

**Comment by dkrat7** [comment]
Another suggestion:
- ability to disable color gradient (to use fixed color)

**Comment by Wojtek17** [comment]
Agreed, the conditional formatting for non-numeric is much needed.

**Comment by lindsay-kk** [comment]
hi, just wanna push this thread since i'm in need of this function too

**Comment by sandeepr43** [comment]
+1

**Comment by joshstackintegrated** [comment]
This would be awesome.
 +1 for "is null"/ "is not null" check

**Comment by tmnvanderberg** [comment]
+9999

**Comment by sanjana-skyflow** [comment]
+1 Conditional formatting for non-numeric columns

**Comment by nsidorenko01** [comment]
+1 Conditional formatting for non-numeric columns

**Comment by ketulgupta1995** [comment]
+1

**Comment by Rusp0** [comment]
+++
Conditional fomratting for non-numeric columns very need for me

**Comment by cj-aki-riisio** [comment]
Would be nice to have the color of a bar / line changed based of another column/value as well

**Comment by enescabri** [comment]
This is surely much needed, thank you all. 

2. The ability to toggle highlighting for an entire row instead of just an individual cell.

**Reply by enescabri** [reply]
Actually, 
we should set conditional formatting for not only integer values. It can be string.

**Comment by avitaltwingo** [comment]
A much needed improvement, would really like to see that happen

**Comment by chungsen** [comment]
a feature that is much needed. If able to conditionally format based on columns not shown on table would be even better.

**Comment by BrayNarvi** [comment]
+1

**Comment by leandrorusso** [comment]
+1

**Comment by sam-hieken** [comment]
+1, it'd also be nice if conditional formatting didn't automatically disable cell bars on other columns

**Comment by PittOpn** [comment]
+1

**Comment by etadelta222** [comment]
+1

**Comment by artrusx** [comment]
+1
It seems to me that this is a simple and obvious improvement, is it really that difficult to add?

**Comment by MarcoFranchin96** [comment]
+1 it's would be a great improvement, it should be develop

**Comment by Anna-Tokar** [comment]
+1

**Comment by XyNG4** [comment]
+1

**Comment by Minimalistikaa** [comment]
- ability to disable color gradient (to use fixed color)
- The ability to apply conditional formatting to non-numeric columns (specifically text and boolean columns).
- The ability to toggle highlighting for an entire row instead of just an individual cell.

When?

**Comment by udalshou-ss** [comment]
+1

**Comment by unappcio** [comment]
+1

**Comment by mboujaha** [comment]
+1

**Comment by sam-hieken** [comment]
If anyone's still looking for a solution to the first issue, happy to say I discovered a great workaround for conditional formatting on non-numeric columns (though unfortunately it's quite hacky).

Basically, you can take advantage of the fact that the table will render columns as HTML if "Render columns in HTML format" is enabled. You can create a calculated column in Superset (or in a view through your database) based on your non-numeric column that conditionally checks values and wraps them in an HTML tag you can style. For example:


<img width="811" alt="Screenshot 2025-02-03 at 7 58 40 PM" src="https://github.com/user-attachments/assets/6c97853b-2164-4bf6-b3d7-c62d4b9f3899" />


Of course we can use a CSS class to handle repetitive styling through the CSS editor in the dashboard...


<img width="551" alt="Screenshot 2025-02-03 at 8 25 33 PM" src="https://github.com/user-attachments/assets/1047db00-72e0-404b-97b5-fb054c6bf715" />


And here's the result:


<img width="432" alt="Screenshot 2025-02-03 at 8 30 40 PM" src="https://github.com/user-attachments/assets/7c0d9420-43ae-4a68-89d9-e71c070e6367" />


Again, not ideal, but at least you also get a lot of flexibility in how it looks. Keep in mind you can just use the original column for dashboard filters so the HTML isn't included in the select options.

**Important Note:** Just to clarify, `level_pretty` is a calculated column (I don't believe this trick will work with metrics, but I'd need to do more testing). The column "Level" in the table I showed is `level_pretty`.

**Reply by isaiaslafon** [reply]
How do I get the chart data? I can create an extra column with the rgb values, but I can't applied. Is there a way in the SQL Expresion to use CSS directly and not HTML? I'm new to apache superset.

**Reply by sam-hieken** [reply]
> How do I get the chart data? I can create an extra column with the rgb values, but I can't applied. 

Not sure exactly what you're asking, but it sounds like maybe you selected the original column to display in the chart instead of the calculated column. Make sure you select the column under the "Saved" tab rather than the "Simple" tab when adding a column in the chart editor. Another possibility is you didn't check the "Render columns in HTML format" box in the "Customize" tab of the editor.

> Is there a way in the SQL Expresion to use CSS directly and not HTML? I'm new to apache superset.

To the best of my knowledge, no. Wouldn't recommend it anyway since the only thing you'd probably want to change between each option is the color, might as well follow DRY and add a class for your common styles in the Live CSS Editor like I did above.

**Reply by isaiaslafon** [reply]
Thx for answer me! Are you in the slack server or on discord to talk? You only submit screenshots, and I don't get how do you get the class name and column data because I don't know your table structures fields internal names. It would be really helpful  more detail data. THX in advance!! I'm not a native speaker, sorry.

**Reply by isaiaslafon** [reply]
I need to change in every cell the color in base of a complex condition that depends on the current date, I also generate an extra column with the color value in hexa, But when using the calculated column style doe not work.

**Reply by isaiaslafon** [reply]
I see the filed is level_pretty and you alter log_level_cell, in the result you have a Level column affected. I you can supply information about the example column name Metric Key, label difference and if you applying to a metric, the same calculated column or a dimension normal column.

**Reply by sam-hieken** [reply]
No worries! Just pinged you on Slack so we can figure out what's going wrong

**Comment by NacerKROUDIR** [comment]
++1
The ability to apply conditional formatting to non-numeric columns is an essential feature in my opinion.

**Comment by paulobitar** [comment]
Your solution is a good workaround but this need to be a feature in superset.

**Reply by enescabri** [reply]
This would be a huge impact on customer side.

**Comment by sam-hieken** [comment]
Yeah I totally agree this is an essential feature but unfortunately I doubt it'll be implemented any time soon given this discussion has been active for so long without being acted on.

**Comment by aryanbhat-fl** [comment]
Is there any feature so that the condition formatting should be applied to the whole row instead of the specified column ?? 

Would love this feature

**Comment by smile2008** [comment]
+1

**Comment by maxQanalytics** [comment]
+1

**Comment by hroche-scene** [comment]
+1

**Comment by DanielKwon01** [comment]
+1

**Comment by MrHassanMurtaza** [comment]
+1

**Comment by Dilaksha** [comment]
yes this is a limtation that I would like to get fixed too.

**Comment by lvxhnat** [comment]
+1

**Comment by stockarea** [comment]
Also to conditionally check other column.
I have 2 columns and their value.. If one of them is greater than the other, i will be needing colour coding based on comparing the 2 values.

**Reply by Scregneugneu** [reply]
> Also to conditionally check other column. I have 2 columns and their value.. If one of them is greater than the other, i will be needing colour coding based on comparing the 2 values.

Want to push for this feature too.
It would be really usefull if we can use other column for conditional formatting, for example in a table chart with an error color scheme when "SUM(Open) < SUM(Send)"

**Comment by endimonan** [comment]
Hello,  if nobody is already working on it, I can pick it up and open a PR. If there’s a preferred scope or approach, let me know.

**Comment by schreibfheler** [comment]
I am in urgent need of having conditional formatting based on datetime columns. E.g. highlight rows that have a date in the past.

**Comment by endimonan** [comment]
I'll check. If it's viable, I'll submit a PR and make it available for members to review.

**Comment by endimonan** [comment]
Quick update after going through the code. The two things originally asked here actually shipped in 6.1.0. Conditional formatting works on text and boolean columns in the Table chart (#34330, #36338), and you can highlight the whole row by picking “entire row” under Formatting column inside the rule popover (#34762, #35897). The fixed color request from the comments is covered too, there’s a “Use gradient” checkbox you can untick. A few caveats: this is all on the classic Table chart. The AG Grid table still only offers numeric columns, and on the Pivot Table string rules only color the header cells. What’s still genuinely missing is datetime support (like highlighting rows with a date in the past, as asked above) and is null checks on non boolean columns.

**Comment by paulobitar** [comment]
Endi,
thanks for the update. I looked for the option "entire row" but I couldn't
find it. Only the graph dimensions are listed on "Formating Collumn" .I
confirmed that Superset is update (6.1). Do I have to enable any flags on
config ?

[image: image.png]
Paulo Bitar Jr
Analista de Sistemas
@Brasília - DF
Mas sempre com Alma em Belém-PA


*Antes de falar, escute. Antes de escrever, pense.Antes de gastar, ganhe.
Antes de rezar, perdoe. Antes de desistir, tente.*
*Muito Importante*
Se você pretende redirecionar este e-mail, por favor, apague todos os NOMES
e ENDEREÇOS que nele aparecem. Outra opção de segurança é endereçá-lo no
Cco ou Bcc. Desta forma você estará protegendo a mim, seus amigos e a você
mesmo. Eu, juntamente com a campanha contra a propagação de vírus
agradecemos sinceramente.


Em qui., 20 de ago. de 2026 às 05:14, Endi Monan ***@***.***>
escreveu:

**Reply by endimonan** [reply]
Hi Paulo,

No feature flag is needed. Just make sure you’re using the classic Table chart.

In the formatting rule, select Background color or Text color under Formatting object. You should then see Entire row under Formatting column.

If Cell bar is selected, the Entire row option won’t be available.

If you still don’t see it, could you share a screenshot of the full formatting window?

**Reply by plaes** [reply]
This is actually quite tricky to find. UI should be more intuitive.

**Comment by paulobitar** [comment]
Thanks for replying ! I was looking at the wrong field of the popup ! Sorry
!
Paulo Bitar Jr
Analista de Sistemas
@Brasília - DF
Mas sempre com Alma em Belém-PA


*Antes de falar, escute. Antes de escrever, pense.Antes de gastar, ganhe.
Antes de rezar, perdoe. Antes de desistir, tente.*
*Muito Importante*
Se você pretende redirecionar este e-mail, por favor, apague todos os NOMES
e ENDEREÇOS que nele aparecem. Outra opção de segurança é endereçá-lo no
Cco ou Bcc. Desta forma você estará protegendo a mim, seus amigos e a você
mesmo. Eu, juntamente com a campanha contra a propagação de vírus
agradecemos sinceramente.


Em seg., 24 de ago. de 2026 às 06:48, Endi Monan ***@***.***>
escreveu:

**Comment by stockarea** [comment]
is someone picking it? for checking strings and checking against other values?

**Comment by rusackas** [comment]
@SBIN2010 was looking at a lot of this, and may have added context. This is an old discussion, partially done, and as much as I'm tempted to close it due to age, there's still more to be done here to bring AG Grid up to parity, and we'd welcome the PRs. Additive features aren't really controversial, and standardization is encouraged, so I'd say anyone/everyone is welcome to open PRs, and we can take things piece by piece.

**Comment by plaes** [comment]
Usability improvement for the "Color the whole row" feature is sitting in #43692.

**Reply by SBIN2010** [reply]
Hi @plaes !
Someone has already added a flag like that; you can see it in this PR- #34762.
And it was changed to the implementation that is now in the PR - #35897
If you look at this PR- #35897, it contains the designer's recommendations and their approval.

**Reply by SBIN2010** [reply]
Perhaps we could simply add a tooltip to avoid duplicating functionality?

**Reply by plaes** [reply]
Ah ok.. I initially had hard time finding the `entire row` entry because I deal with tables which have 20-30 columns..

And I actually started this as a "preparation" to my other todo-list: https://github.com/apache/superset/discussions/43646

**Reply by SBIN2010** [reply]
Excellent additions. I am currently working on the third item on your list. If you have any ideas regarding the interface for implementing that third point, I would be happy to take a look at them.
